docs/how-to/performance_guidelines.rst

```diff
@@ -1,125 +1,102 @@
+
+On the other hand, device level optimization focuses on the device as a whole,
+aiming at keeping all multiprocessors busy by executing enough kernels
+concurrently. Both multiprocessor and device levels of optimization are crucial for achieving maximum
 performance. They work together to ensure efficient utilization of the
-resources of the GPU, from the individual multiprocessors to the device as a
+GPU resources, ranging from individual multiprocessors to the device as a
 whole.
 
 .. _memory optimization:
-
-Memory optimization
-===================
+Memory throughput optimization
+===============================
 
 The first step in maximizing memory throughput is to minimize low-bandwidth
-data transfers. This involves reducing data transfers between the host and the
-device, as these have lower bandwidth than transfers between global memory and
-the device.
-
-Additionally, data transfers between global memory and the device should be
-minimized by maximizing the use of on-chip memory: shared memory and caches.
-Shared memory acts as a user-managed cache, where the application explicitly
-allocates and accesses it. A common programming pattern is to stage data from
-device memory into shared memory. This involves each thread of a block loading
-data from device memory to shared memory, synchronizing with all other threads
-of the block, processing the data in shared memory, synchronizing again if
-necessary, and writing the results back to device global memory.
+data transfers between the host and the device.
 
-For some applications, a traditional hardware-managed cache is more appropriate
-to exploit data locality. On devices of certain compute capabilities, the same
-on-chip memory is used for both L1 and shared memory, and the amount dedicated
-to each is configurable for each kernel call.
-
-Finally, the throughput of memory accesses by a kernel can vary significantly
-depending on the access pattern for each type of memory. Therefore, the next
-step in maximizing memory throughput is to organize memory accesses as
-optimally as possible. This is especially important for global memory accesses,
-as global memory bandwidth is low compared to available on-chip bandwidths and
-arithmetic instruction throughput. Thus, non-optimal global memory accesses
-generally have a high impact on performance.
+Additionally, maximize the use of on-chip memory, that is, shared memory and caches, and minimize transfers with global memory. Shared memory acts as a user-managed cache explicitly allocated and accessed by the application. A common programming pattern is to stage data from device memory into shared memory. The staging of data from the device to shared memory involves:
+1. Each thread of a block loading data from device memory to shared memory.
+2. Synchronizing with all other threads of the block.
+3. Processing the data stored in shared memory.
+4. Synchronizing again if necessary.
+5. Writing the results back to the device global memory.
```

Comment:
These don't display on new lines. Please make the changes.

Reply:
Fixed.